tests/complex/run.sh

```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 
-TEST_DIR="$(realpath $(dirname $0))"
+# shellcheck disable=SC2034
 TESTNAME=""
+TEST_DIR="$(realpath "$(dirname "${0}")")"
```

Comment:
Ugh, I did not even know that this works. Do the brackets take precedence over the quotes?